code/game/objects/structures/kitchen_spike.dm

```diff
@@ -84,57 +110,51 @@
 	if(..())
 		return
 	if(src.occupied)
-		if(src.meattype == MEATTYPE_MONKEY && src.skintype == SKINTYPE_MONKEY)
-			if(src.skin >= 1)
-				src.skin--
-				new /obj/item/stack/sheet/animalhide/monkey(src.loc)
-				user << "<span class='notice'>You remove the hide from the monkey.</span>"
-			else if(src.meat > 1)
-				src.meat--
-				new /obj/item/weapon/reagent_containers/food/snacks/meat/slab/monkey(src.loc )
-				usr << "<span class='notice'>You remove some meat from the monkey.</span>"
-			else if(src.meat == 1)
-				src.meat--
-				new /obj/item/weapon/reagent_containers/food/snacks/meat/slab/monkey(src.loc)
-				usr << "<span class='notice'>You remove the last piece of meat from the monkey.</span>"
-				src.icon_state = "spike"
-				src.occupied = 0
-		else if(src.meattype == MEATTYPE_ALIEN && src.skintype == SKINTYPE_ALIEN)
-			if(src.skin >= 1)
-				src.skin--
-				new /obj/item/stack/sheet/animalhide/xeno(src.loc)
-				user << "<span class='notice'>You remove the hide from the alien.</span>"
-			else if(src.meat > 1)
-				src.meat--
-				new /obj/item/weapon/reagent_containers/food/snacks/meat/slab/xeno(src.loc )
-				usr << "<span class='notice'>You remove some meat from the alien.</span>"
-			else if(src.meat == 1)
-				src.meat--
-				new /obj/item/weapon/reagent_containers/food/snacks/meat/slab/xeno(src.loc)
-				usr << "<span class='notice'>You remove the last piece of meat from the alien.</span>"
-				src.icon_state = "spike"
-				src.occupied = 0
-		else if(src.meattype == MEATTYPE_BEAR && src.skintype == SKINTYPE_BEAR)
-			if(src.skin >= 1)
-				src.skin--
-				new /obj/item/clothing/head/bearpelt(src.loc)
-				user << "<span class='notice'>You remove the hide from the bear.</span>"
-			else if(src.meat > 1)
-				src.meat--
-				new /obj/item/weapon/reagent_containers/food/snacks/meat/slab/bear(src.loc )
-				usr << "<span class='notice'>You remove some meat from the bear.</span>"
-			else if(src.meat == 1)
-				src.meat--
-				new /obj/item/weapon/reagent_containers/food/snacks/meat/slab/bear(src.loc)
-				usr << "<span class='notice'>You remove the last piece of meat from the bear.</span>"
-				src.icon_state = "spike"
-				src.occupied = 0
+		if (src.meat > 1)
+			switch(src.meattype)

```

Comment:
you can probably address the [x]type comments I made by making the MEATTYPE and SKINTYPE defines paths; see __DEFINES/pipe_construction.dm to see what I mean by that, and game/machinery/pipe/construction.dm for how it was implemented